tests/conftest.py

```diff
@@ -115,18 +124,77 @@ def clean_serveradmin(filters):
     Query(filters).delete().commit()
 
 
+def clean_aws(vm_hostname):
+    def _get_instance_status():
+        response = ec2.describe_instances(
+            Filters=[
+                {
+                    'Name': 'instance-state-code',
+                    'Values': [
+                        str(AWS_RETURN_CODES['pending']),
+                        str(AWS_RETURN_CODES['running']),
+                        str(AWS_RETURN_CODES['shutting-down']),
+                        str(AWS_RETURN_CODES['terminated']),
+                        str(AWS_RETURN_CODES['stopping']),
+                        str(AWS_RETURN_CODES['stopped']),
+                    ]
+                },
+            ],
+            InstanceIds=[obj['aws_instance_id']],
+            DryRun=False)['Reservations'][0]['Instances'][0]['State']['Code']
+
+        return int(response)
+
+    try:
+        obj = Query({'hostname': vm_hostname}, ['aws_instance_id']).get()
+    except DatasetError:  # No object to clean up
+        return
+
+    if obj['aws_instance_id']:
+        timeout = 120
+        ec2 = boto3.client('ec2')
+        try:
+            ec2.stop_instances(
+                InstanceIds=[obj['aws_instance_id']], DryRun=False
+            )
+        except ClientError as e:
+            pass  # Not running
+        for retry in range(timeout):
+            instance_status = _get_instance_status()
+            if AWS_RETURN_CODES['stopped'] == instance_status:
+                break
+            sleep(1)
+
+        ec2.terminate_instances(InstanceIds=[obj['aws_instance_id']])
+        for retry in range(timeout):
+            instance_status = _get_instance_status()
+            if AWS_RETURN_CODES['terminated'] == instance_status:
```

Comment:
what happens if it isnt terminated in 120s?

Reply:
Usually a machine is terminate within seconds, so 120 should be more than enough.